src/util/ConfigManager/ConfigOptionProxy.h

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (March of 2023, schlegea@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <absl/strings/str_cat.h>
+
+#include <concepts>
+#include <stdexcept>
+#include <type_traits>
+
+#include "util/ConfigManager/ConfigOption.h"
+#include "util/Exception.h"
+
+namespace ad_utility {
+namespace detail {
+/*
+@brief Implementation of proxy/reference to a `ConfigOption`. Also saves the
+type of value, that the configuration option holds. Needed, because without a
+`ConfigOptionProxy` and `ConstConfigOptionProxy`, people could easily discard
+the `const` from a `const ConfigOptionProxy` by copying it, to have non const
+access to the referenced config option.
+
+@tparam T The type of value, the later given config option will hold.
+@tparam ConfigOptionType The kind of config option, this proxy will reference
+to. Must be `ConfigOption`, or `const ConfigOption`.
+*/
+template <typename T, typename ConfigOptionType>
+requires ad_utility::isTypeContainedIn<T, ConfigOption::AvailableTypes> &&
+         (std::same_as<std::decay_t<ConfigOptionType>, ConfigOption> ||
+          std::same_as<std::decay_t<ConfigOptionType>, const ConfigOption>)
+class ConfigOptionProxyImplementation {
+  ConfigOptionType* option_ = nullptr;
```

Comment:
Don't specify the `nullptr` here. Because then tools can warn if it is not set in one of the constructors.

Reply:
Done, see [2fe7028](https://github.com/ad-freiburg/qlever/pull/1055/commits/2fe7028c03e48a178a0dd102524106a35f649366).